modules/biobank/php/analysis.class.inc

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+/**
+ * Specimen Analysis Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Analysis instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\biobank;
+
+/**
+ * Specimen Analysis Class
+ * This class contains an instance of a database handling that is needed to
+ * permanently store and retrieve Specimen Analaysis instances.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Entity
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+class Analysis extends Process
+{
+    /**
+     * Persistent Instance variables.
+     *
+     * These variables are directly mapped to the fields of the database schema
+     * that describe a specimen entity.
+     *
+     * @access private
+     */
+    private $methodId; //int
+
+    /**
+     * Sets the ID of the specimen's method
+     *
+     * @param int $methodId Method ID to be set
+     *
+     * @return void
+     */
+    public function setMethodId(int $methodId) : void
```

Comment:
These setters and getters don't seem to have much point other than enforcing types, which could be detected by phan if the `@var` annotation was used.. if you insist on setters, then returning `$this` instead of void would allow you to write fluent interfaces.

Reply:
I think that this point can be discussed in our meetings regarding loris entities. @ridz1208 @xlecours @johnsaigle